src/main/java/com/google/sps/servlets/ScheduledInterviewServlet.java

```diff
@@ -83,36 +101,87 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   // Send the request's contents to Datastore in the form of a new ScheduledInterview object.
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String requestedInterviewerId = request.getParameter("interviewer");
-    String requestedIntervieweeId = request.getParameter("interviewee");
-    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
-    String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+    String intervieweeEmail = userService.getCurrentUser().getEmail();
+    String intervieweeId = userService.getCurrentUser().getUserId();
+
     // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
     // replaced with this hashcode.
-    if (userId == null) {
-      userId = String.format("%d", userEmail.hashCode());
+    if (intervieweeId == null) {
+      intervieweeId = String.format("%d", intervieweeEmail.hashCode());
     }
-    // The default key for a scheduledInterview being stored in datastore
-    long defaultKey = -1;
-    if ((!requestedInterviewerId.equals(userId)) && (!requestedIntervieweeId.equals(userId))) {
-      response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+
+    // TODO: Put this null id check in a separate method to call from both Get and Post.
```

Comment:
???

Reply:
Done. Created a method getUserId() that handles the null case.